ksql-engine/src/test/java/io/confluent/ksql/structured/SchemaKTableTest.java

```diff
@@ -162,4 +186,62 @@ public void testFilter() throws Exception {
                       initialSchemaKTable);
   }
 
+  @Test
+  public void testGroupBy() {
+    String selectQuery = "SELECT col0, col1, col2 FROM test1;";
+    PlanNode logicalPlan = planBuilder.buildLogicalPlan(selectQuery);
+    initialSchemaKTable = new SchemaKTable(logicalPlan.getTheSourceNode().getSchema(), kTable,
+        ksqlTable.getKeyField(), new ArrayList<>(), false,
+        SchemaKStream.Type.SOURCE, functionRegistry, new MockSchemaRegistryClient());
+
+    Expression col1Expression = new DereferenceExpression(
+        new QualifiedNameReference(QualifiedName.of("TEST1")), "COL1");
+    Expression col2Expression = new DereferenceExpression(
+        new QualifiedNameReference(QualifiedName.of("TEST1")), "COL2");
+    KsqlTopicSerDe ksqlTopicSerDe = new KsqlJsonTopicSerDe();
+    Serde<GenericRow> rowSerde = ksqlTopicSerDe.getGenericRowSerde(
+        initialSchemaKTable.getSchema(), null, false, null);
+    List<Expression> groupByExpressions = Arrays.asList(col2Expression, col1Expression);
+    SchemaKGroupedStream groupedSchemaKTable = initialSchemaKTable.groupBy(
+        Serdes.String(), rowSerde, groupByExpressions);
+
+    assertThat(groupedSchemaKTable, instanceOf(SchemaKGroupedTable.class));
+    assertThat(groupedSchemaKTable.getKeyField().name(), equalTo("TEST1.COL2|+|TEST1.COL1"));
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldGroupKeysCorrectly() {
```

Comment:
I think it would really help in tests like this to list out the givens and expectation in words. Its really hard to follow what you are trying to test from the code, with all the mocks and intermediate steps involved.